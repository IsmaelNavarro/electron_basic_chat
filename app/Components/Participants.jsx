import React from 'react';
import TimeAgo from 'react-timeago';
import PropTypes from 'prop-types';

export default class Participant extends React.Component {

    static defaultProps = {
        client: null
    }

    static propTypes = {
        client: PropTypes.object.isRequired
    }

    constructor(props) {
        super(props);
        this.state = {
            participants: []
        };

        props.client.on('participants', this.onClientParticipants);
    }

    onClientParticipants = (participants) => {
        this.setState({
            participants: participants
        });
    }

    render() {
        const { participants } = this.state;
        return (
            <div className="pane pane-sm sidebar">
                <ul className="list-group">
                    {participants.map((user, i) => (
                        <li key={i} className="list-group-item">
                            <div className="media-body">
                                <strong><span className="icon icon-user"></span>&nbsp;{user.name}</strong>
                                <p>Joined <TimeAgo date={user.time} /> </p>
                            </div>
                        </li>
                    ))}
                </ul>
            </div>
        );
    }

}