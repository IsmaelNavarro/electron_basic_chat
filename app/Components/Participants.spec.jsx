import * as React from 'react';
import Client from '../Service/Client';
import Participants from './Participants';
import TestRenderer from 'react-test-renderer';

describe('Participants', () => {
    it('matches previously saved snapshot', () => {
        const items = [{
            name: 'Jon',
            time: new Date(2012, 2, 12, 5, 5, 5, 5)
        }],
            client = new Client(),
            component = TestRenderer.create(<Participants client={client} />);

        component.getInstance().onClientParticipants(items);
        expect(component.toJSON()).toMatchSnapshot();
    });
}); 