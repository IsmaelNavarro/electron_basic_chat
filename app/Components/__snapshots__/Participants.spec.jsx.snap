// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants matches previously saved snapshot 1`] = `
<div
  className="pane pane-sm sidebar"
>
  <ul
    className="list-group"
  >
    <li
      className="list-group-item"
    >
      <div
        className="media-body"
      >
        <strong>
          <span
            className="icon icon-user"
          />
          Â 
          Jon
        </strong>
        <p>
          Joined 
          <time
            dateTime="2012-03-12T04:05:05.005Z"
            title="2012-03-12 04:05"
          >
            6 years ago
          </time>
           
        </p>
      </div>
    </li>
  </ul>
</div>
`;
